### PR TITLE
[IA-1857] add error reporting

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-1857

Manually tested in console
Screen Shot 2020-08-07 at 9 08 17 AM

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: errorReporting/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#Testing Locally in sbt console
+
+This is an example on how to test Google2 methods locally. In this example, we are testing the 
+GoogleStorageInterpreter.getObjectMetadata method. Before this, we have downloaded a credential JSON 
+file with which we can create a GoogleStorageService object.
```
Comment (kyuksel) [inline errorReporting/README.md]: The description above needs to be updated for the example below.

File: errorReporting/src/main/scala/org/broadinstitute/dsde/workbench/errorReporting/ErrorReportingInterpreter.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      .setMessage(msg)
+      .setServiceContext(serviceContext)
+      .setContext(ErrorContext.newBuilder().setReportLocation(sourceLocation))
+      .build()
+
+    F.delay(client.reportErrorEvent(projectName, errorEvent))
+  }
+
+  override def reportError(t: Throwable): F[Unit] = {
+    val stackTraceWriter = cats.effect.Resource.make {
```
Comment (kyuksel) [inline errorReporting/src/main/scala/org/broadinstitute/dsde/workbench/errorReporting/ErrorReportingInterpreter.scala]: Remove `cats.effect` for consistency with other `Resource` usages?

**Review by kyuksel — APPROVED** [review]
LGTM 👍  with minor comments inline

**Review by rtitle — APPROVED** [review]
Code looks fine -- was wondering how we decide to put things in a different module. Since this uses a Google SDK I could also see adding it to the `google2` module.

**Comment by Qi77Qi** [comment]
@rtitle I put it as a separate module since metrics is a separate module as well, also google2 has way too many dependencies which aren't needed for some apps...I'm thinking maybe we should break up google2 into individual modules as well, and have `model` or potentially a separate module for defining shared data types

**Pull request merged**